source/modulo_new_core/include/modulo_new_core/communication/PublisherHandler.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "modulo_new_core/communication/PublisherInterface.hpp"
+
+namespace modulo_new_core::communication {
+
+template<typename PubT, typename MsgT>
+class PublisherHandler : public PublisherInterface {
+public:
+  PublisherHandler(PublisherType type, std::shared_ptr<PubT> publisher);
+
+  void publish(const MsgT& message) const;
+
+private:
+  std::shared_ptr<PubT> publisher_;
```

Comment:
At the PublisherHandler level, the `DataT` is still needed right? And when we `add_publisher` we use the data type to create both the handler and the message pair 
```suggestion
  std::shared_ptr<PubT<DataT>> publisher_;
```

Reply:
sorry, I guess it should actually be `MsgT`

Reply:
No, you cannot do <PubT<MsgT>>...

Reply:
If you could to that, it would simplify a lot haha..maybe it gets clearer if I push the implementation and the tests?

Reply:
Yeah I need to rethink some things at the top level. Here it makes sense that `PubT` refers to the specific publisher type (like `rclcpp::Publisher<std_msgs::msg::Bool>` or whatever), but the `PubT` we have in the top-level ComponentInterface was not supposed to work like that.

So for now I would suggest we remove `PubT` as a top-level template in `ComponentInterface` and just use the `PublisherType type` as a construction argument that the derived classes set